### DISCUSSION TITLE
databend-query build failed on macOS

### DISCUSSION THREAD

**Discussion by yuliyu123** [databendlabs/databend#5860] [General]
**Summary**
when i build this project, error reported below:
error: failed to run custom build command for `common-protos v0.1.0

how i can fix this issue?

**Comment by wubx** [comment]
Maybe follow step help:
```
which protoc
goto the path:
sudo mv protoc proctoc-bak
```

maket setup -d

**Comment by yuliyu123** [comment]
useless, maybe has relations with apple M1 chip.

**Comment by wubx** [comment]
> 

M1 works fine. 
Can you provide more detail about your compile process and more output log.

https://github.com/wubx/dba-in-databend/blob/main/install/compile_databend_in_china/complie-databend-in-china.md

**Comment by everpcpc** [comment]
maybe run `./scripts/setup/dev_setup.sh`  would help

**Reply by Maricaya** [reply]
It solved my problem, thanks a lot!

**Comment by yuliyu123** [comment]
macOS 12.3.1
m1 chip

config:
`❯ rustc --version
rustc 1.63.0-nightly (cd282d7f7 2022-05-18)

❯ protoc --version
libprotoc 3.19.4`

make build-release 
`bash ./scripts/build/build-release.sh
Build(RELEASE) start...
   Compiling databend-query v0.1.0 (/Users/looperX/Devlop/rust/databend/query)
error: linking with `cc` failed: exit status: 1
  |
  = note: "cc" "-m64" "-arch" "x86_64" "
...
st/databend/target/release/deps/databend_query-6ac85c55ade712f3" "-Wl,-dead_strip" "-nodefaultlibs"
  = note: ld: warning: ignoring file /opt/homebrew/Cellar/xz/5.2.5/lib/liblzma.dylib, building for macOS-x86_64 but attempting to link with file built for macOS-arm64
          Undefined symbols for architecture x86_64:
            "_lzma_auto_decoder", referenced from:
                xz2::stream::Stream::new_auto_decoder::hc1bac2a8128d00b2 in databend_query-6ac85c55ade712f3.xz2-30bfe1099b286f37.xz2.db5c5b59-cgu.0.rcgu.o.rcgu.o
            "_lzma_code", referenced from:
                _$LT$async_compression_issue_150_workaround..codec..xz2..decoder..Xz2Decoder$u20$as$u20$async_compression_issue_150_workaround..codec..Decode$GT$::decode::ha0ee39bdeb2ca0ec in databend_query-6ac85c55ade712f3.opendal-b2c28a61fbdd27f8.opendal.53b1e0fa-cgu.0.rcgu.o.rcgu.o
                _$LT$async_compression_issue_150_workaround..codec..xz2..decoder..Xz2Decoder$u20$as$u20$async_compression_issue_150_workaround..codec..Decode$GT$::finish::h3059cc4c4328f7f1 in databend_query-6ac85c55ade712f3.opendal-b2c28a61fbdd27f8.opendal.53b1e0fa-cgu.0.rcgu.o.rcgu.o
            "_lzma_end", referenced from:
                core::ptr::drop_in_place$LT$core..option..Option$LT$opendal..io_util..compress..DecompressDecoder$GT$$GT$::h5977790bb0bc0b99 in databend_query-6ac85c55ade712f3.databend_query-f11ae4486bf6a909.databend_query.6e20be77-cgu.0.rcgu.o.rcgu.o
                core::ptr::drop_in_place$LT$opendal..io_util..compress..DecompressCodec$GT$::he553629866272f69 in databend_query-6ac85c55ade712f3.databend_query-f11ae4486bf6a909.databend_query.6e20be77-cgu.1.rcgu.o.rcgu.o
                core::ptr::drop_in_place$LT$opendal..io_util..compress..DecompressCodec$GT$::he553629866272f69 (.llvm.8757268150426916830) in databend_query-6ac85c55ade712f3.databend_query-f11ae4486bf6a909.databend_query.6e20be77-cgu.3.rcgu.o.rcgu.o
                _$LT$opendal..io_util..compress..DecompressCodec$u20$as$u20$async_compression_issue_150_workaround..codec..Decode$GT$::reinit::hc7f931c381840cd2 in databend_query-6ac85c55ade712f3.opendal-b2c28a61fbdd27f8.opendal.53b1e0fa-cgu.0.rcgu.o.rcgu.o
                core::ptr::drop_in_place$LT$xz2..stream..Stream$GT$::hbac4ac641354a22d (.llvm.4828314909460718141) in databend_query-6ac85c55ade712f3.xz2-30bfe1099b286f37.xz2.db5c5b59-cgu.0.rcgu.o.rcgu.o
                xz2::stream::Stream::new_auto_decoder::hc1bac2a8128d00b2 in databend_query-6ac85c55ade712f3.xz2-30bfe1099b286f37.xz2.db5c5b59-cgu.0.rcgu.o.rcgu.o
          ld: symbol(s) not found for architecture x86_64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
          

error: could not compile `databend-query` due to previous error
All done...

`

**Reply by wubx** [reply]
It‘s unfortunate. but it happens. 
```
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
meaning lost golang++, c++

> maybe run `./scripts/setup/dev_setup.sh` would help

if you need help please join our stack,  I'm glad to help you. 

https://link.databend.rs/join-slack

**Comment by ygf11** [comment]
Maybe `brew install xz` or `brew install lzip` can help,

Since `lzma` lib is need here:
```
Undefined symbols for architecture x86_64:
"_lzma_auto_decoder", referenced from:
xz2::stream::Stream::new_auto_decoder::hc1bac2a8128d00b2 in databend_query-6ac85c55ade712f3.xz2
```

ref:
https://tukaani.org/xz/
https://www.nongnu.org/lzip/

**Reply by Xuanwo** [reply]
Nice catch!

**Comment by FourSpaces** [comment]
I found that before successfully manufacturing the data terminal, you need to install protobuf and thrift and configure the environment variables of protobuf.
However, this part is not mentioned in the document

**Comment by zhexuany** [comment]
I met the same error which resolved by linking running `brew link protobuf`
